frontend/react/style-guide.md

```diff
@@ -642,17 +657,24 @@ src
   Notes for use:
   Filter out unnecessary props when possible. Also, use [prop-types-exact](https://www.npmjs.com/package/prop-types-exact) to help prevent bugs.
 
-  ```jsx
-  // good
+  ```tsx
+    // bad
   function Component(props) {
-    const { irrelevantProp, ...relevantProps } = props;
-    return <WrappedComponent {...relevantProps} />
+    return <WrappedComponent {...props} />
   }
 
-  // bad
-  function Component(props) {
-    return <WrappedComponent {...props} />
+  interface Props {
+    irrelevantProp: string,
+    firstRelevantProp: string,
+    secondRelevantProp: string
+  };
+
+  // good
+  function Component(props: Props) {
+    const { irrelevantProp, ...relevantProps } = props;
+    return <WrappedComponent {...relevantProps} />
   }
```

Comment:
i dont sure in which cases we need send a irrelevant prop. 

it can be that we need send specific props to the parent and child components, but i think that the destructuring can doing in the declaration of the component. 

```suggestion
  interface Props {
    parentProp: string,
    firstChildProp: string,
    secondChildProp: string
  };

  // good
  function Component({ parentProp, ...childProps}: Props) {
    return <WrappedComponent {...childProps} />
  }
```

what do you think about it?

Reply:
I really like this way better. I will apply it